### DISCUSSION TITLE
CS_Tools install is failing

### DISCUSSION THREAD

**Discussion by RandallHoffpauir** [thoughtspot/cs_tools#79] [Q&A] [answered]
Hi,
I trying to get started with cs_tools.  I have installed python version 3.12.0.  When I try to install cs_tools I get errors. I am not sure how to get around this issue. cs_tools is not installed. I have attached the error log.
[cs_tools-bootstrap-error-tmpoo__qws2.log](https://github.com/thoughtspot/cs_tools/files/13309211/cs_tools-bootstrap-error-tmpoo__qws2.log)  Any help would be greatly appreciated. Thanks.

**Comment by boonhapus** [comment, marked as answer]
Hey @RandallHoffpauir - we don't yet support python 3.12 . 

We have a reliance on `pendulum` which has **[yet to add](https://github.com/sdispater/pendulum/issues/753)** their **[support for 3.12](https://github.com/sdispater/pendulum/issues/696)** . In our next update, I'll restrict python version between `>= 3.7 , < 3.12`

**Reply by RandallHoffpauir** [reply]
Thanks. That was the issue. I installed python 3.11.6 and the cs_tools installed just fine!
